12 create runner class

LuisFelipe-PE opened the pull request:
This pull request introduces the runner_object and the test file for the runner object. Is also includes the modifications for the config files to run in win32.

LuisFelipe-PE committed:
Runner object created

Script for run the models with an initial test case

LuisFelipe-PE committed:
Runner object and test modified

Runner object: included set_inputs and get_outputs functions (empty)/removed the result line and return -->none
Test runner: modified with structure

LuisFelipe-PE committed:
Fixed for typos and test

Modified the scripts for the typecheck and test validations

LuisFelipe-PE committed:
Update pyproject toml and env

Updated the pyproject toml to correctly install the required packages.

LuisFelipe-PE committed:
Fix type error pywanda

Pywanda not properly typed

samvanderzwan committed:
Downgrade version number of python to work with pywanda

samvanderzwan committed:
Fixed version number

samvanderzwan committed:
changed to run on windows for pywanda

samvanderzwan committed:
Put build package  back, it will faill

samvanderzwan committed:
removed lien

samvanderzwan committed:
set path to win32

samvanderzwan committed:
Removed building of package and surfacing of failing tests

LuisFelipe-PE committed:
Fixed Lint warnings

LuisFelipe-PE committed:
Fixed lint warnings v2

LuisFelipe-PE committed:
Fixed Lint warnings v3

vanmeerkerk commented on .github/workflows/ci.yml:
```diff
@@ -25,103 +25,45 @@ jobs:
 
   test:    
-    runs-on: ubuntu-latest
+    runs-on: windows-latest
```
Why do you need to change the ci files? These are run by the continuous integration environment in github and I believe that runs on ubuntu not windows?

vanmeerkerk replied:
Checked with @samvanderzwan; windows is required for pywanda as it can not run on a linux machine.

vanmeerkerk commented on src/pywandahydra/entities/runner_object.py:
```diff
@@ -1,10 +1,82 @@
-    pass
+        model = pywanda.WandaModel(str(Path(self.model_path, self.model_name)), self.bin)
+        # Store all the input information
+        print("Save model input")
```
Why would you save the model input when you performed no changes to the model?

LuisFelipe-PE replied:
Fixed, the save input will be included in the parameter file itself

vanmeerkerk commented on src/pywandahydra/entities/runner_object.py:
```diff
@@ -1,10 +1,82 @@
+
+        Returns:
+        --------
+        result: list of strings
```
The Returns are not correct; the method currently returns None.

LuisFelipe-PE replied:
Changed the result  to none

vanmeerkerk commented on src/pywandahydra/entities/runner_object.py:
```diff
@@ -1,10 +1,82 @@
+        try:
+            model.run_steady()
+        except RuntimeError as e:
+            print(e)
```
These errors need to be written to a log. Is there a logging module in the package?

@samvanderzwan

vanmeerkerk commented on unit_test/entities/test_runner_object.py:
```diff
@@ -0,0 +1,20 @@
+from pywandahydra.entities.runner_object import RunnerObject
+
+
+class MyTest(unittest.TestCase):
```
The test should have a clear name and follow the convention ClassNameTest in this case RunnerObjectTest

vanmeerkerk commented on unit_test/entities/test_runner_object.py:
```diff
@@ -0,0 +1,20 @@
+class MyTest(unittest.TestCase):
+    """Class for testing of the running object."""
+
+    def test_runner_object(self) -> None:
```
The test should follow the Arrange-Act-Assert structure for example see the below code:

``
     def test_set_physical_properties_missing_key(self) -> None:
          """Test the set_physical_properties method."""
          # arrange
          physical_properties_dict = {
              PROPERTY_DIAMETER: 0.5,
              PROPERTY_LENGTH: 2000.0,
          }
  
          # act
          with self.assertRaises(ValueError) as cm:
              self.asset.set_physical_properties(physical_properties=physical_properties_dict)  # act
  
          # assert
          self.assertIsInstance(cm.exception, ValueError)
          self.assertEqual(
              str(cm.exception), f"Property {PROPERTY_ROUGHNESS} is missing in physical_properties"
          )
``

vanmeerkerk commented on unit_test/simulation/Sewage_transient_WPS.rtf:
The RTF file is not necessary and I believe you can put them on the gitignore list

vanmeerkerk commented on unit_test/simulation/Sewage_transient_WPS.wdd:
Maybe this can also be put in the gitignore list? Only keep the .csv, .wdi .wdx .wdo .wps

vanmeerkerk commented on unit_test/test_main.py:
```diff
@@ -28,6 +28,9 @@ def tearDown(self) -> None:
         reload(logging)
         return super().tearDown()
 
+    def setUp(self) -> None:
```
Why did you make this test?

vanmeerkerk reviewed (changes requested):
See comments

LuisFelipe-PE committed:
Initial review requirements

Include additional files in gitignore, modified runner_object as required and change name in test_runner_object
